app/scripts/analysis/tools/powercalc/PowercalcOneTGUI.service.coffee

```diff
@@ -0,0 +1,174 @@
+'use strict'
+
+BaseService = require 'scripts/BaseClasses/BaseService.coffee'
+
+module.exports = class PowerCalc_OneTGUI extends BaseService
+    
+  initialize: ->
+    @distanceFromMean = 5
+
+
+  extract: (data, variable) ->
+    tmp = []
+    for d in data
+      tmp.push +d[variable]
+    tmp
+
+  getRightBound: (middle,step) ->
+    return middle + step * @distanceFromMean
+
+  getLeftBound: (middle,step) ->
+    return middle - step * @distanceFromMean
+
+  sort: (values) ->
+    values.sort (a, b) -> a-b
+
+  getVariance: (values, mean) ->
+    temp = 0
+    numberOfValues = values.length
+    while( numberOfValues--)
+      temp += Math.pow( (parseInt(values[numberOfValues]) - mean), 2 )
+
+    return temp / values.length
+
+  getSum: (values) ->
+    values.reduce (previousValue, currentValue) -> parseFloat(previousValue) + parseFloat(currentValue)
+
+  getGaussianFunctionPoints: (std, mean, variance, leftBound, rightBound) ->
+    data = []
+    for i in [leftBound...rightBound] by 1
+      data.push
+        x: i
+        y:(1 / (std * Math.sqrt(Math.PI * 2))) * Math.exp(-(Math.pow(i - mean, 2) / (2 * variance)))
+    data
+
+  getMean: (valueSum, numberOfOccurrences) ->
+    valueSum / numberOfOccurrences
+
+  getZ: (x, mean, standardDerivation) ->
+    (x - mean) / standardDerivation
+
+  getWeightedValues: (values) ->
+    weightedValues= {}
+    data= []
+    lengthValues = values.length
+    for i in [0...lengthValues] by 1
+      label = values[i].toString()
+      if(weightedValues[label])
+        weightedValues[label].weight++
+      else
+        weightedValues[label]={weight :1,value :label}
+        data.push(weightedValues[label])
+    return data
+
+  getRandomNumber: (min,max) ->
+    Math.round((max-min) * Math.random() + min)
+
+  getRandomValueArray: (data) ->
+    values = []
+    length = data.length
+    for i in [1...length]
+      values.push data[Math.floor(Math.random() * data.length)]
+    return values
+
+  drawNormalCurve: (mean0, mean_in1, variance_in1, sigma_in1, alpha_in) ->
+    margin = {top: 20, right: 20, bottom: 20, left:20}
+    width = 500 - margin.left - margin.right
+    height = 500 - margin.top - margin.bottom
+
+    container = d3.select("#One_TGUI_graph")
+    container.select("svg").remove()
+
+    svg = container.append('svg')
```

Comment:
shouldn't do any rendering in the controllers, this should be moved into a directive. see how it's done in [Clustering](https://github.com/SOCR/SOCRAT/blob/dev/app/scripts/analysis/tools/Cluster/ClusterVizDir.directive.coffee#L26)